chcemvediet/apps/obligees/tests/test_models.py

```diff
@@ -1,369 +1,672 @@
+
+    def test_history_records_changes_to_gender_field(self):
+        oblg = self._create_obligee(gender=Obligee.GENDERS.MASCULINE)
+        oblg.gender = Obligee.GENDERS.FEMININE
+        oblg.save()
+        history = oblg.history.all()
+        self.assertEqual([h.gender for h in history], [Obligee.GENDERS.FEMININE, Obligee.GENDERS.MASCULINE])
+
+    def test_history_records_changes_to_ico_field(self):
+        oblg = self._create_obligee(ico=u'12345678')
+        oblg.ico = u'87654321'
+        oblg.save()
+        history = oblg.history.all()
+        self.assertEqual([h.ico for h in history], [u'87654321', u'12345678'])
+
+    def test_history_records_changes_to_street_field(self):
+        oblg = self._create_obligee(street=u'123 Westside')
+        oblg.street = u'123 Eastside'
+        oblg.save()
+        history = oblg.history.all()
+        self.assertEqual([h.street for h in history], [u'123 Eastside', u'123 Westside'])
+
+    def test_history_records_changes_to_city_field(self):
+        oblg = self._create_obligee(city=u'Winterfield')
+        oblg.city = u'Springfield'
+        oblg.save()
+        history = oblg.history.all()
+        self.assertEqual([h.city for h in history], [u'Springfield', u'Winterfield'])
+
+    def test_history_records_changes_to_zip_field(self):
+        oblg = self._create_obligee(zip=u'12345')
+        oblg.zip = u'99999'
+        oblg.save()
+        history = oblg.history.all()
+        self.assertEqual([h.zip for h in history], [u'99999', u'12345'])
+
+    def test_history_records_changes_to_iczsj_field(self):
+        neighbourhood1 = self._create_geounit(id=u'12312')
+        neighbourhood2 = self._create_geounit(id=u'12345')
+        oblg = self._create_obligee(iczsj=neighbourhood1)
+        oblg.iczsj = neighbourhood2
+        oblg.save()
+        history = oblg.history.all()
+        self.assertEqual([h.iczsj_id for h in history], [u'12345', u'12312'])
+
+    def test_history_records_changes_to_emails_field(self):
+        oblg = self._create_obligee(emails=u'agency@example.com')
+        oblg.emails = u'agency@example.com, another@example.com'
+        oblg.save()
+        history = oblg.history.all()
+        self.assertEqual([h.emails for h in history], [u'agency@example.com, another@example.com', u'agency@example.com'])
+
+    def test_history_records_changes_to_latitude_field(self):
+        oblg = self._create_obligee(latitude=44.11388888888889)
+        oblg.latitude = -23.451
+        oblg.save()
+        history = oblg.history.all()
+        self.assertEqual([h.latitude for h in history], [-23.451, 44.11388888888889])
+
+    def test_history_records_changes_to_longitude_field(self):
+        oblg = self._create_obligee(longitude=-85.03111111111112)
+        oblg.longitude = 13.8321
+        oblg.save()
+        history = oblg.history.all()
+        self.assertEqual([h.longitude for h in history], [13.8321, -85.03111111111112])
+
```

Comment:
@viliambalaz By sa zislo doplnit aj testy, ze `tags` a `groups` sa nedrzia v historii (kedze to je m2m relacia cez inu tabulku). Teda `test_history_does_not_record_changes_to_tags_field`, kde zmenis tag a assertnes, ze `h.tags` raisne exception, ze history nema `tags`.

Reply:
:ok: